stack.py

```diff
@@ -0,0 +1,18 @@
+from linkedlist import LinkedList
+
+
+class Stack:
+    def __init__(self, values=None):
+        self.storage = LinkedList(values)
+
+    def push(self, val):
+        self.storage.push(val)
+
+    def pop(self):
+        try:
+            return self.storage.pop()
+        except ValueError:
+            raise ValueError("The stack is empty.")
+
+    def __len__(self):
+        return self.storage._length
```

Comment:
When a method/property is prefixed with an `_` it means it is private and is for internal use within the owning object only.  Here you are using the private property `_length` of the LinkedList within the stack.  Change this to use only public methods/properties.